schedulearn-frontend/src/components/Suggestions/CreateSuggestion.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import { EventModal } from "../Calendar/EventModal";
+import { SuggestionForm } from "./SuggestionForm";
+import ManagedTeamsSelect from "../../server-components/ManagedTeamsSelect";
+import UsersInTeamList from "../../server-components/UsersInTeamList";
+import UserContext from "src/api-services/UserContext";
+import CreateNewSuggestion from "src/api-services/api-contract/CreateNewSuggestion";
+
+export interface NewSuggestion {
+  topicId?: number;
+  suggesterId?: number;
+  suggesteeId?: number;
+}
+
+export interface CreateSuggestionProps {
+  onSubmit: (newSuggestion: CreateNewSuggestion) => void; 
+}
+
+interface CreateSuggestionState {
+  newSuggestion?: NewSuggestion;
+  isSuggestionModalOpen: boolean;
+  currentTeamId?: number;
+}
+
+export class CreateSuggestion extends React.Component<CreateSuggestionProps, CreateSuggestionState> {
+  public state: CreateSuggestionState = {
+    isSuggestionModalOpen: false,
+  }
+  
+  handleModalClose = (): void => {
+    this.setState({
+      isSuggestionModalOpen: false,
+    });
+  }
+
+  handleEventSubmit = (topicId: number): void => {
+    if (topicId === undefined)
```

Comment:
If this can actually ever be undefined, then topicId type should be 
topicId?: number (or topicId: number | undefined).

But my guess is that topicId is never actually undefined?

Reply:
Yep, removed the unnecessary check